### ISSUE TITLE
[Visualforce] VfUnescapeEl not resolved on multi argument Javascript

### ISSUE THREAD

**Issue by elerno** [pmd/pmd#5398]
**Affects PMD Version:** 
7.6.0 (included in Salesforce Code Analyzer)

**Rule:**
[VfUnescapeEl](https://docs.pmd-code.org/pmd-doc-7.6.0/pmd_rules_visualforce_security.html#vfunescapeel)

**Description:**
Wrapping in JSENCODE or JSINHTMLENCODE is not preventing PDM from triggering VfUnescapeEl. I haven't found documentation, posts or bug reports with a similar case so, while I think this is a false positive, there might be something I'm missing here. Please advise!

**Code Sample demonstrating the issue:**

```VISUALFORCE
<div class="pre-wrap" title="Sync"><a href="javascript:if('{!t.inSub}'!='true'){alert('This institution is not part of your subscription - it will be updated but with minimal data');}sync('{! JSENCODE(t.id) }', '{! JSENCODE(t.uid) }', '{!t.inSub}', '{!JSENCODE(t.name)}', '{!JSENCODE(t.address1)}', '{!JSENCODE(t.address2)}', '{!JSENCODE(t.city)}', '{!JSENCODE(t.state)}');">Sync/Merge</a></div>
```

**Expected outcome:**

PMD reports a violation at line ..., but I think that's wrong. Is this a false positive?

**Running PMD through:** *
CLI (sf scanner 4.7.0)

Thanks!

**Comment by jsotuyod** [comment]
Mmmm... According to your sample it seems to be correct at first glance...

`'{!t.inSub}'` is effectively unescaped both times it's being used.

**Comment by elerno** [comment]
Thanks for your answer @jsotuyod!

I'm afraid that `'{!t.inSub}' ` is a boolean type, and if I unescape it, I get the following error when trying to deploy to the Org:

`=== Deploy Errors
PROJECT PATH                                        ERRORS                                            
──────────────────────────────────────────────────  ──────────────────────────────────────────────────
force-app/main/default/pages/Account_Matching.page  Incorrect argument type for function 'JSENCODE()'.`

Still, PMD complains about all arguments, not just the booleans. Please see the results CSV:

[CodeAnalyzerGeneral_2024-12-11.csv](https://github.com/user-attachments/files/18116707/CodeAnalyzerGeneral_2024-12-11.csv)
